app/main.py

```diff
@@ -1,13 +1,31 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    def format_linter_error(error):
+        return {
+            "line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"
+        }
+
+
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    def format_single_linter_file(file_path, errors):
+        return {
+            "errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "failed" if errors else "passed"
+        }
+
+
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    def format_linter_report(linter_report):
```

Comment:
The nested function definition `def format_linter_report(linter_report):` is redundant. The outer function definition already serves the purpose. Consider removing this line.